### PR TITLE
tls: add timestamp and sni info for a session

### PR DESCRIPTION
**Pull request by smititelu**
#### Pre-Submission Checklist

- [X] Commit message has the format required by CONTRIBUTING guide
- [X] Commits are split per component (core, individual modules, libs, utils, ...)
- [X] Each component has a single commit (if not, squash them into one commit)
- [X] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [X] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

### CODE REVIEW

**Comment by henningw** [comment]
@smititelu Thanks for the pull request, did a quick review. It would be great if you could also check for eventual errors at the snprintf call for the time stamp. It might for some reasons fails, as the TLS module might see error conditions during heavy load.

**Comment by smititelu** [comment]
Updated PR. Will print "N/A" for timestamp too, if snprintf call fails.

File: src/modules/tls/tls_rpc.c [hunk 1]
```diff
@@ -132,14 +135,29 @@ static void tls_list(rpc_t* rpc, void* c)
...
+					timestamp.tm_min, timestamp.tm_sec) < 0) {
+				timestamp_s[0] = 'N';
+				timestamp_s[1] = '/';
+				timestamp_s[2] = 'A';
+				timestamp_s[3] = '\0';
+			}
+			rpc->struct_add(handle, "dsdsdsd",
 					"id", con->id,
+					"timestamp", timestamp_s,
 					"timeout", timeout,
```
Comment (oej) [inline src/modules/tls/tls_rpc.c:150]: Can we get SNI here as well?

**Comment by oej** [comment]
Great addition.

File: src/modules/tls/tls_rpc.c (hunk 1 above)
Comment (smititelu) [reply src/modules/tls/tls_rpc.c:150]: you mean to print sni right after the "id"?!

Comment (oej) [reply src/modules/tls/tls_rpc.c:150]: Sounds like a good place.

Comment (smititelu) [reply src/modules/tls/tls_rpc.c:161]: :+1: updated PR accordingly

**Comment by oej** [comment]
So what do we do if the SNI doesn't match any known SIP domain we handle? Like you have a wildcard cert in your organisation and someone connects to the wrong DNS name?

**Comment by smititelu** [comment]
Will print that wrong requested name.

**Comment by henningw** [comment]
@smititelu thanks for the updates. @oej This is a generic question, right? Then it should be probably discussed on the mailing list and not in this PR.

**Comment by oej** [comment]
It is related to getting SNI available in the config - but could of course be discussed on the mailing list as well. I was wondering what Stefan thought about this since he added better SNI support :-)

**Comment by oej** [comment]
But it is not a blocker for this PR, not a requirement.

**Comment by oej** [comment]
> Will print that wrong requested name. E.g.

Right, but we should propably think about closing early if that's the case or find a way to handle it in the routing script properly. But at that point, we've already accepted the SIP message. Let's continue this discussion outside of this PR.

**Comment by miconda** [comment]
Thanks, merging!

**Pull request merged**